sdk/spring/spring-messaging-azure-eventhubs/src/main/java/com/azure/spring/eventhubs/core/processor/DefaultEventHubsNamespaceProcessorFactory.java

```diff
@@ -143,4 +150,20 @@ public boolean removeListener(Listener listener) {
         return this.listeners.remove(listener);
     }
 
+    /**
+     * Set the token credential resolver.
+     * @param tokenCredentialResolver The token credential resolver.
+     */
+    public void setTokenCredentialResolver(AzureCredentialResolver<AzureTokenCredentialProvider> tokenCredentialResolver) {
+        this.tokenCredentialResolver = tokenCredentialResolver;
+    }
+
+    /**
+     * Set the default Azure credential.
+     * @param defaultAzureCredential The default Azure Credential.
+     */
+    public void setDefaultAzureCredential(DefaultAzureCredential defaultAzureCredential) {
+        this.defaultAzureCredential = defaultAzureCredential;
```

Comment:
same as above noted. setter better not set the default value.

Reply:
It is by design.